tests/20-rdfc-verify.js

```diff
@@ -21,62 +21,45 @@ const {match} = endpoints.filterByTag({
 
 describe('ecdsa-rdfc-2019 (verify)', function() {
   describe('ecdsa-rdfc-2019 (verifiers)', function() {
-    let issuers;
-    before(async function() {
-      const {match} = endpoints.filterByTag({
-        tags: [...tags],
-        property: 'issuers'
-      });
-      // Uses 'Digital Bazaar' as default issuer to issue a verifiable
-      // credential for the verifier tests.
-      issuers = match.get(issuerName).endpoints;
-    });
     // this will tell the report
     // to make an interop matrix with this suite
     this.matrix = true;
     this.report = true;
     this.rowLabel = 'Test Name';
     this.columnLabel = 'Verifier';
     this.implemented = [];
+    let testVectors = new Map();
+    before(async function() {
+      testVectors = await issueTestData({
+        credential: credentials.verify.document,
+        suite: 'ecdsa-rdfc-2019',
+        keyTypes
+      });
+    });
     for(const [name, {endpoints: verifiers}] of match) {
       for(const verifier of verifiers) {
         const {
-          supportedEcdsaKeyTypes: verifierSupportedEcdsaKeyTypes
+          supportedEcdsaKeyTypes: verifierKeyTypes
         } = verifier.settings;
-        const keyTypes = verifierSupportedEcdsaKeyTypes.join(', ');
+        const keyTypes = verifierKeyTypes.join(', ');
         // add implementer name and keyTypes to test report
         this.implemented.push(`${name}: ${keyTypes}`);
         describe(`${name}: ${keyTypes}`, function() {
-          const credentials = [];
-          beforeEach(async function() {
-            // use each issuer to create test data for verify suite
-            for(const issuer of issuers) {
-              const {
-                supportedEcdsaKeyTypes: issuerSupportedEcdsaKeyTypes
-              } = issuer.settings;
-              // issue vc for each supported key type
-              for(const verifierSupportedEcdsaKeyType of
-                verifierSupportedEcdsaKeyTypes) {
-                if(issuerSupportedEcdsaKeyTypes.includes(
-                  verifierSupportedEcdsaKeyType)) {
-                  // add each vc to resulting test data
-                  const issuedVc = await createInitialVc({
-                    issuer,
-                    vc: vcTestData.verify.document
-                  });
-                  credentials.push(issuedVc);
-                }
-              }
-            }
+          let supportedVectors = [];
+          before(function() {
+            // filter the test data to only include VC signed with
+            // keyTypes the verifier supports
+            supportedVectors = verifierKeyTypes.map(
+              (keyType = '') => testVectors.get(keyType.toUpperCase()));
```

Comment:
Noting this was a `beforeEach`, now it is a single `before`.

I cant quite tell - why is the `before` needed here at all?

Reply:
So the before statement above this one creates all of the test data at once, then each implementer tells us which keyTypes they support, so we filter the testData created above so we don't test an implementer against a keyType they don't support. It's actually redundant here as this suite only supports `p-256`, but that could change. I might add that a subsequent PR will change the way this all works anyways as it turns out there are further issues with the way `supportedEcdsaKeyTypes` works.